triggers/service/authentication.md

```diff
@@ -0,0 +1,95 @@
+# Design for Trigger Service Authentication/Authorization
+
+## Goals
+
+- Be compatible with an OAuth2 authorization code grant
```

Comment:
👏 so you decided to go with the code grant.

Reply:
At least I want to support this as it seems like the most natural model for the trigger service. I do believe that you can fit a client credentials grant in there if you want to. In this case, the middleware would act as follows:

/login would somehow check that the user is authorized to interact with the trigger service. This is deliberately outside of the client credentials grant so I don’t think there is a standard solution to this. One option would be to just require the user to authenticate as some admin user with whatever authentication mechanism you have. If that is successful, the middleware makes the request to get the token and then proceeds as the middleware for the authorization code grant.